kernel/arch/dreamcast/include/dc/syscalls.h

```diff
@@ -0,0 +1,394 @@
+    \ingroup system_calls
+
+    This function checks if a queued command has completed.
+
+    \param  id              The request id (>=0).
+    \param  status          The pointer to four 32-bit integers to 
+                            receive status information.
+
+    \retval 0               Request not found.
+    \retval 1               Request is still being processed.
+    \retval 2               Request completed successfully.
+    \retval 3               Stream type command is in progress.
+    \retval 4               GD syscalls are busy.
+
+    \sa syscall_gdrom_send_command(), syscall_gdrom_run_commands()
+*/
+int syscall_gdrom_check_command(uint32_t id, void *status);
+
+/** \brief   Process queued GDROM commands.
+    \ingroup system_calls
+
+    This function starts processing queued commands. This must be 
+    called a few times to process all commands.
+
+    \sa syscall_gdrom_send_command(), syscall_gdrom_check_command()
+*/
+void syscall_gdrom_run_commands(void);
+
+/** \brief   Abort a queued GDROM command.
+    \ingroup system_calls
+
+    This function tries to abort a previously queued command.
+
+    \param  id              The request id (>=0) to abort.
+
+    \return                 0 on success, or non-zero on
+                            failure.
+*/
+int syscall_gdrom_abort_command(uint32_t id);
+
+/** \brief   Sets/gets the sector mode for read commands.
+    \ingroup system_calls
+
+    This function sets/gets the sector mode for read commands.
+
+    \param  status          The pointer to a struct of four 32 bit integers 
+                            containing new values, or to receive the old 
+                            values.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+*/
+int syscall_gdrom_sector_mode(void *mode);
+
+/** \brief   Clear user defined vectors.
+    \ingroup system_calls
+
+    This function clears all the user defined syscall vectors.
+
+    \return                 0
+*/
+int syscall_misc_init(void);
+
+/** \brief   Register/Clear a user defined vector
+    \ingroup system_calls
+
+    This function sets/clears the handler for one of the seven user defined
+    vectors. Setting a handler is only allowed if it not currently set.
+
+    \param  vector          The vector number (1-7).
+    \param  handler         The pointer to handler function, or NULL to
+                            clear.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+*/
+int syscall_misc_setvector(uint8_t super, uintptr_t handler);
+
+/** \brief   Setup GDROM DMA callback
+    \ingroup system_calls
+
+    This function sets up DMA transfer end callback for 
+    CMD_DMAREAD_STREAM_EX (see cdrom.h).
+
+    \param  callback        The function to call upon completion of the DM.
+    \param  param           The data to pass to the callback function.
+*/
+void syscall_dma_callback(uintptr_t callback, void *param);
```

Comment:
Missed "gdrom" in func name here and down.

Reply:
Good catch!